libraries/splunk_app.rb

```diff
@@ -126,13 +174,80 @@ def create_splunk_directory(path)
         dir.group(node['splunk']['group'])
         dir.mode('0755')
         dir.run_action(:create)
-        new_resource.updated_by_last_action(dir.updated_by_last_action?)
+        new_resource.updated_by_last_action(true) if dir.updated_by_last_action?
+      end
+
+      def download_and_install # rubocop:disable PerceivedComplexity, CyclomaticComplexity
+        return if new_resource.url.nil? || new_resource.url.empty?
+
+        expected_version = CernerSplunk::AppVersion.new new_resource.version
+        installed_version = CernerSplunk::AppVersion.new @current_resource.version
+
+        return if expected_version.type && expected_version == installed_version
+        Chef::Log.warn "Expected version not specified for #{new_resource.app}." unless expected_version.type
```

Comment:
I'm confused by the `unless expected_version.type` here as well for the same reason as my confusion on the previous line.